Marketplace/controllers/api/apiProductController.js

```diff
@@ -1,11 +1,77 @@
-const db = require("../../database/models");
-const entidad = db.Producto;
+const productoRepository = require("../../repositories/productoRepository");
+const categoriaRepository = require("../../repositories/categoriaRepository");
 
 module.exports = {
-	listado: (req, res) => {
-
+	listado: async (req, res) => {
+		let listado = null;
+		listado = await productoRepository.ObtenerTodas();
+		// *** PRODUCTOS ***
+		let productos = [];
+		listado.map(n => {
+			let aux=[]
+			n.imagenes.map(m => aux.push(m.ruta));
+			productos.push({
+				id: n.id,
+				nombre: n.nombre,
+				descripcion: n.descripcion,
+				categoria: n.categoria.nombre,
+				imagenes: aux,
+				url: "/api/productos/" + n.id,
+			});
+		});
```

Comment:
Me parece que aca no estamo susando bien la funcion map, yo lo haria asi:
```suggestion
		let productos = listado.map(n => {
			return {
				id: n.id,
				nombre: n.nombre,
				descripcion: n.descripcion,
				categoria: n.categoria.nombre,
				imagenes: n.imagenes.map(m => m.ruta),
				url: "/api/productos/" + n.id,
			}
		});
```

Reply:
Muy de acuerdo. No se me había ocurrido hacerlo de esa manera. Lo vas a ver así cuando actualice el pull.